src/controllers/users.js

```diff
@@ -0,0 +1,32 @@
+const mongoose = require('mongoose');
+const { User } = require('../models/user');
+
+const addUserSubscription = async (req, res) => {
+    try {
+        const followUserId = mongoose.Types.ObjectId(req.params.id);
+        const user = await User.findByIdAndUpdate(req.user.id, {
+            $push: {
+                followedUsers: followUserId,
+            },
+        });
+
+        if (!user) {
```

Comment:
here you are checking user after posting sub to it! you have to check user in  param.